### ISSUE TITLE
SQL error from incorrect table name when changing all table collations

### ISSUE THREAD

**Issue by ibennetch** [phpmyadmin/phpmyadmin#14575]
### Steps to reproduce
1. On a database Operations tab, select from the dropdown to change the collation
2. Check both checkboxes for "Change all tables collations" and "Change all tables columns collations"

The SQL query that's generated has a 0 instead of my table name, causing the query to fail:

![image](https://user-images.githubusercontent.com/1443710/44295950-92ab9780-a281-11e8-9dfb-573c20d079a1.png)

> SQL query: Edit Edit
> 
> ```ALTER TABLE `collationtest`.`0`DEFAULT CHARSET=ascii COLLATE ascii_bin```
> 
> MySQL said: Documentation
> `#1146 - Table 'collationtest.0' doesn't exist`

Tested with current master

**Comment by williamdes** [comment]
@ibennetch The error you reported, I can not reproduce it but I have another one.
Create a view in a database and try to change collation on all tables and columns. 
```sql
SET SQL_MODE = "NO_AUTO_VALUE_ON_ZERO";
SET AUTOCOMMIT = 0;
START TRANSACTION;
SET time_zone = "+00:00";


CREATE TABLE `event` (
  `name` varchar(20) COLLATE utf8mb4_et_0900_ai_ci DEFAULT NULL,
  `date` date DEFAULT NULL,
  `type` varchar(15) COLLATE utf8mb4_et_0900_ai_ci DEFAULT NULL,
  `remark` varchar(255) COLLATE utf8mb4_et_0900_ai_ci DEFAULT NULL
) ENGINE=InnoDB DEFAULT CHARSET=utf8mb4 COLLATE=utf8mb4_et_0900_ai_ci;

CREATE TABLE `owners` (
`owner` varchar(20)
);




CREATE TABLE `pet` (
  `name` varchar(20) COLLATE utf8mb4_hr_0900_ai_ci DEFAULT NULL,
  `owner` varchar(20) COLLATE utf8mb4_hr_0900_ai_ci DEFAULT NULL,
  `species` varchar(20) COLLATE utf8mb4_hr_0900_ai_ci DEFAULT NULL,
  `sex` char(1) COLLATE utf8mb4_hr_0900_ai_ci DEFAULT NULL,
  `birth` date DEFAULT NULL,
  `death` date DEFAULT NULL
) ENGINE=InnoDB DEFAULT CHARSET=utf8mb4 COLLATE=utf8mb4_hr_0900_ai_ci;

INSERT INTO `pet` (`name`, `owner`, `species`, `sex`, `birth`, `death`) VALUES
('Puffball', 'Diane', 'hamster', 'f', '1999-03-30', NULL);

CREATE VIEW `owners`  AS  select distinct `pet`.`owner` AS `owner` from `pet` ;
COMMIT;
```

![image](https://user-images.githubusercontent.com/7784660/49087502-78596080-f257-11e8-96c3-2231b72fddf5.png)

```#1347 - 'menagerie.owners' is not BASE TABLE```

**Comment by kartik1000** [comment]
@ibennetch, Hey is the issue some table specific or based on some particular properties of table because when I tried reproducing it, it worked fine for me on `world` database of demo server
![screenshot 2019-02-24 at 1 04 19 pm](https://user-images.githubusercontent.com/43112279/53296423-cd941680-3834-11e9-9d36-d9bba45b9124.png)

**Comment by kartik1000** [comment]
@williamdes, Hey also your sql query is causing collation error on my local server. Is it working fine for you still?
![screenshot 2019-02-24 at 12 56 01 pm](https://user-images.githubusercontent.com/43112279/53296441-25328200-3835-11e9-8990-86df156e0a54.png)

**Comment by williamdes** [comment]
Hey @kartik1000 The collation must be an error, the most important part to reproduce this issue is to have a view in the database :)

**Comment by M-Mommsen** [comment]
Same problem here - a view stops the alter operations on a database! (latest "stable" from git)
Only workaround so far -> delete the views, alter the collation of the database, recreate the views ... which is not very practical!

**Comment by williamdes** [comment]
Duplicate of #15283

**Comment by williamdes** [comment]
Fixed by fd7f2397d36fdcbae7123ef078c536e8e197ffc4
